lib/gears/matcher.lua

```diff
@@ -100,10 +100,19 @@ end
 --
 -- @param o The object.
 -- @tparam[opt=nil] table rules The rules to check. List with "rule", "rule_any",
---  "except" and "except_any" keys.
+--  "except" and "except_any" keys. If no rules are provided, the default set
+--  will be used.
 -- @treturn table The list of matched rules.
 function matcher:matching_rules(o, rules)
+    rules = rules or select(2, next(self._matching_rules))
```

Comment:
Well, that one is tricky. The old `awful.rules` (before the v4.3 release) could be "massaged" into being re-usable with non-client objects by passing a different rules array to that function (blueyed did). This makes zero sense anymore. If someone wants to do that, then having multiple matcher objects is much more sane.

I will change "default" to "at random", but in 99% of the case, there's only one.

Reply:
Uhm. So why can one provide different names for these rules anyway? If I understand you correctly, it makes no sense to provide different names, because then one ends with "I will pick at random which rules to apply".

Put differently: If I were to remove all `name` arguments from `gears.matcher` (and magically fixed all callers not to provide one), what would break?